### PR TITLE
Privacy notice table

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fidesplus/issues/743

### Code Changes

* [x] Use react-table to render the privacy notices table
* [x] Custom cells to render the various different data types
* [x] Sorting by column
* [x] A bare bones individual page for a privacy notice
* [x] Cypress tests
* [x] Lots of tweaking to match the designs
* [x] Enable toggle (not hooked up yet, but scaffolding is there)

### Steps to Confirm

* Run fidesplus (I used `nox -s fides_env -- prod_edge`)
* You'll need some privacy notices! I've been using a JSON file of privacy notices based off of the design and posting it to the endpoint via `http POST localhost:8080/api/v1/privacy-notice 'Authorization: Bearer ey...[copied from network tab after logging in]' < privacyNotices.json` 

<details>
<summary>privacyNotices.json</summary>

```json
[
  {
    "name": "Essential",
    "description": "Notify the user about data processing activities that are essential to your services functionality. Typically consent is not required for this.",
    "consent_mechanism": "necessary",
    "regions": ["eu_fr", "eu_ie"],
    "data_uses": ["provide.service"],
    "enforcement_level": "system_wide"
  },
  {
    "name": "Functional",
    "description": "This is for data processing activities that enhance the capability or features of your site but may not be strictly necessary.",
    "consent_mechanism": "opt_in",
    "regions": ["eu_fr", "eu_ie"],
    "data_uses": ["improve.system"],
    "enforcement_level": "system_wide"
  },
  {
    "name": "Analytics",
    "description": "Ensures you are correctly notifying the user about your advertising practices and for appropriate locations, collecting the users consent preferences.",
    "consent_mechanism": "opt_in",
    "regions": ["eu_fr", "eu_ie"],
    "data_uses": ["collect"],
    "enforcement_level": "system_wide"
  },
  {
    "name": "Advertising",
    "description": "Ensures you are correctly notifying the user about your advertising practices and for appropriate locations, collecting the users consent preferences.",
    "consent_mechanism": "opt_in",
    "regions": ["eu_fr", "eu_ie"],
    "data_uses": ["advertising.first_party.contextual"],
    "enforcement_level": "system_wide"
  },
  {
    "name": "Data Sales",
    "description": "Provide opt-out consent for the use of data in ways that may be considered “data sales” under US state privacy regulations.",
    "consent_mechanism": "opt_out",
    "regions": ["us_ca", "us_co"],
    "data_uses": ["advertising.third_party.personalized"],
    "enforcement_level": "frontend"
  }
]

```
</details>

* The table will only show notices for which the data use is linked to a system. So now go into a system, edit its privacy declaration to give it one of the data uses you have a privacy notice for (i.e. `provide.service` for "Essential" if you're using the `privacyNotices.json` file above
* That notice should now show up in the table
  * You can also work around this by commenting out the `param` field in `privacy-notices.slice.ts` which sends the flag `system_applicable=true`. Without that flag, you don't have to link the system to the data use which should make things a little easier
* Try clicking on the column headers to sort the columns
* Clicking on a table row should bring you to a WIP page for the notice

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [x] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [x] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [x] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [x] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### Description Of Changes

https://user-images.githubusercontent.com/24641006/230476776-d1a4afeb-96dd-4903-9f3c-ecf438a06824.mov

Regions overflow

image

### CODE REVIEW

**Comment by allisonking** [comment]
A few in the weeds questions for you @mfbrown 

Do we have a mapping of how we'd like regions displayed? the backend uses i.e. `us_ca` and `eu_ie` which I just made all caps in the UI. the design has some variants i.e. the US ones have `US-` prepended by the EU ones do not. I can make a mapping, but that would also mean if we add anymore we'd have to update the code
image

I want to double check the enable/disable behavior—clicking the toggle doesn't fire off the API call right away, right? The user has to click "save" first? If so, I've got two questions there:
* Do we want the confirmation modal for when someone toggles something off? It's not saved yet, so I could see that being a little misleading
* There's a slightly grey background if a notice is disabled. Should this be for if the notice is actually disabled (as in it is disabled in the backend) or as soon as a user marks it as disabled (might not be disabled in the backend yet, just a temporary UI state)

File: clients/admin-ui/src/features/privacy-notices/cells.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+}: CellProps<PrivacyNoticeResponse, boolean>) => (
+  <Switch
+    colorScheme="complimentary"
+    isChecked={value}
+    onChange={() => {
+      /**
+       * It's difficult to use a custom column in react-table 7 since we'd have to modify
+       * the declaration file. However, that modifies the type globally, so our datamap table
+       * would also have issues. Ignoring the type for now, but should potentially revisit
+       * if we update to react-table 8
```
Comment (TheAndrewJackson) [inline clients/admin-ui/src/features/privacy-notices/cells.tsx:72]: I think it's worth investigating upgrading to react-table 8 in the near future. I'm pretty sure it was rewritten in typescript so it should help solve any typescript related issues

**Review by TheAndrewJackson — APPROVED** [review]
Awesome! 🚀 Everything looked good. Just one small comment about react-table 8

File: clients/admin-ui/src/features/privacy-notices/cells.tsx (hunk 1 above)
Comment (allisonking) [reply clients/admin-ui/src/features/privacy-notices/cells.tsx:72]: 👍 I'll make a ticket!

**Comment by mfbrown** [comment]
> A few in the weeds questions for you @mfbrown
> 
> Do we have a mapping of how we'd like regions displayed? the backend uses i.e. `us_ca` and `eu_ie` which I just made all caps in the UI. the design has some variants i.e. the US ones have `US-` prepended by the EU ones do not. I can make a mapping, but that would also mean if we add anymore we'd have to update the code image
> 
> I want to double check the enable/disable behavior—clicking the toggle doesn't fire off the API call right away, right? The user has to click "save" first? If so, I've got two questions there:
> 
> * Do we want the confirmation modal for when someone toggles something off? It's not saved yet, so I could see that being a little misleading
> * There's a slightly grey background if a notice is disabled. Should this be for if the notice is actually disabled (as in it is disabled in the backend) or as soon as a user marks it as disabled (might not be disabled in the backend yet, just a temporary UI state)

Regarding the grey background - this is the hover state of the row, since users click into rows now to get into the details of items. It's not to do with the disabled state.

For the confirmation on disable - We can make a bigger change here and actually remove the Save/Cancel buttons from this table. The only action you can take on this page is to enable or disable, so I think it makes sense to take that action when the toggle is flipped and not require an additional click on save. 

For the locations, I'm ok for now including the `EU-` suffix on the chips like we're doing for US.

**Pull request merged**